crates/query-engine/translation/src/translation/helpers.rs

```diff
@@ -202,6 +223,23 @@ impl<'request> Env<'request> {
             })
     }
 
+    /// Lookup type representation of a type.
+    pub fn lookup_type_representation(
+        &self,
+        typ: &metadata::Type,
```

Comment:
Makes sense. Perhaps we can add a method on type `scalar_type() : Type -> Option<ScalarType>` that we can use on the caller side and it will make this more straightforward.